lib/response.js

```diff
@@ -21,14 +21,18 @@ function Response (req, onEnd, reject) {
 
   this._promiseCallback = typeof reject === 'function'
 
+  let called = false
   const onEndSuccess = (payload) => {
+    called = true
```

Comment:
Didn't add `if (called) return` here as `onEndSuccess` is bound to be called only `once` on `finish`. So adding the condition here it's extraneous.

Reply:
Please add a comment for it